custom_components/acwd/acwd_api.py

```
@@ -228,16 +266,26 @@ def get_usage_data(self, mode='B', date_from=None, date_to=None, str_date=None,
+                headers={'User-Agent': USER_AGENT}
+            )
+            
+            if page_response.status_code == 200:
+                soup = BeautifulSoup(page_response.text, 'html.parser')
+                csrf_input = soup.find('input', {'id': 'hdnCSRFToken'})
+                if csrf_input:
+                    fresh_csrf = csrf_input.get('value', '')
+                    if fresh_csrf:
+                        self.csrf_token = fresh_csrf
+                        logger.debug("Got fresh CSRF token from usage page")
+        except Exception as e:
```

Ruff (0.14.8)

Do not catch blind exception: `Exception`

(BLE001)